core/core-spacemacs-buffer.el

```diff
@@ -828,9 +826,53 @@ list. Return entire list if `END' is omitted."
                   (when (spacemacs-buffer//insert-file-list
                          "Projects:"
                          (spacemacs//subseq (projectile-relevant-known-projects)
-                                     0 list-size))
+                                            0 list-size))
                     (spacemacs//insert--shortcut "p" "Projects:")
-                    (insert list-separator)))))) dotspacemacs-startup-lists))))
+                    (insert list-separator)))))) dotspacemacs-startup-lists)))
+
+(defun spacemacs-buffer//get-buffer-width ()
+  (save-excursion
+    (goto-char 0)
+    (let ((current-max 0))
+      (while (not (eobp))
+        (let ((line-length (- (line-end-position) (line-beginning-position))))
+          (if (< current-max line-length)
+              (setq current-max line-length)))
+        (forward-line 1))
+      current-max)))
+
+(defvar spacemacs-buffer--buttons-position nil
+  "Offset in characters between the edge of the screen and the beginning of the
+ home buffer buttons. Do not try to set this variable.")
+
+(defvar spacemacs-buffer-buttons-recent-lists-offset 25
```

Comment:
This seems to be independent of the actual width of the startup lists, i.e. not really centered. Am I missing something?

In any case this should probably not have `recent` in the name.

Would prefer to move the defvars to the top of the file if possible.